### PR TITLE
Shared String Dictionary ctd.

### PR DESCRIPTION
**Pull request by Tyler-IN**
fix up integration tests

includes some simple docker stuff too

depends on https://github.com/space-wizards/RobustToolbox/pull/1126

### CODE REVIEW

File: Content.IntegrationTests/Tests/EntityTest.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@
 namespace Content.IntegrationTests.Tests
 {
     [TestFixture]
+    [NonParallelizable, SingleThreaded, RequiresThread]
```
Comment (PJB3005) [inline Content.IntegrationTests/Tests/EntityTest.cs:17]: Running integration tests in parallel is something that is 100% desirable in the future due to their relatively high cost currently so I feel this is a crappy bandaid at best.

File: Content.IntegrationTests/Tests/EntityTest.cs [hunk 2]
```diff
@@ -75,6 +76,12 @@ public async Task Test()
                 });
 
             await server.WaitIdleAsync();
+
+            server.Stop();
```
Comment (PJB3005) [inline Content.IntegrationTests/Tests/EntityTest.cs:80]: This is way too boilerplatey and error prone to have to stick at the end of every integration test. I actually tried to make it so this was done automatically with a `[TearDown]` so I would prefer if you figured out why this isn't working instead.

File: Content.IntegrationTests/Tests/HumanInventoryUniformSlotsTest.cs [hunk 3]
```diff
@@ -80,6 +83,19 @@ public async Task Test()
...
+
+            await server.WaitIdleAsync();
+
+            Assert.That(server.IsAlive, Is.False);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Interlocked.MemoryBarrierProcessWide();
```
Comment (PJB3005) [inline Content.IntegrationTests/Tests/HumanInventoryUniformSlotsTest.cs:97]: What?

**Comment by Tyler-IN** [comment]
It was leftover troubleshooting crap. Shouldn't have been committed.

**Comment by PJB3005** [comment]
You removed *some* of the explicit integration instance shutdowns but not all, why?

**Comment by Tyler-IN** [comment]
Cause I didn't revert, I just removed

**Pull request closed without merging**